### PR TITLE
Remove slice patterns.

### PR DESCRIPTION
**Pull request by steveklabnik**
So yeah, this works, but is waaaaay more awkward. I'm 99% sure I got this right. You may or may not want to accept this, but I figured I'd offer it.

This only leaves the associated constant gate.

### CODE REVIEW

**Comment by Johnson-A** [comment]
Slice patterns really do simplify this! I might consider using this method if I want to use stable rust in the near future. The associated constants are only used for Move::NULL and Entry::NULL, there might be a better way to implement them, however.

**Comment by tbu-** [comment]
You could probably get away with enabling `#[feature(test)]` only if you're compiling the test binary:

```
#![cfg_atttr(test, feature(test))]
#[cfg(test)]
extern crate test;
```

That would only leave the `associated_consts` feature gate which could probably also be removed some way.

**Comment by steveklabnik** [comment]
Yup, that'd be the way to handle that with tests. But it's not worth bothering with until the other gates are all gone.

File: src/board.rs [hunk 1]
```diff
@@ -164,14 +164,20 @@ impl Board {
...
-            let mut flags = match promotion {
-                ['q'] => QUEEN_PROM,
-                ['r'] => ROOK_PROM,
-                ['b'] => BISHOP_PROM,
-                ['n'] => KNIGHT_PROM,
+            let mut flags = match promotion[0] {
+                'q' => QUEEN_PROM,
+                'r' => ROOK_PROM,
+                'b' => BISHOP_PROM,
+                'n' => KNIGHT_PROM,
```
Comment (tbu-) [inline src/board.rs:180]: Crashes if promotion has length zero.

Comment (steveklabnik) [inline src/board.rs:180]: ah, good call :/

**Pull request closed without merging**